datafusion/src/physical_plan/functions.rs

```diff
@@ -111,6 +112,11 @@ pub fn return_type(
             utf8_to_int_type(&input_expr_types[0], "character_length")
         }
         BuiltinScalarFunction::Chr => Ok(DataType::Utf8),
+        BuiltinScalarFunction::Coalesce => {
+            // COALESCE has multiple args and they might get coerced, get a preview of this
+            let coerced_types = data_types(input_expr_types, &signature(fun));
```

Comment:
Is there any doc or reference to guide for the common result type or the coercion type?